extra/import_data.py

```diff
@@ -0,0 +1,174 @@
+            vus_export = None
+            while vus_export is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
+                    vus_export = alissa_client.get_inheritance_analyses_variants_export(analysis['id'], export_id)
+                except HTTPError as exception:
+                    pass
+                except json.decoder.JSONDecodeError:
+                    break
+            logger.info('Alissa retrieval completed of: %s' % patient_dn_no)
+            # sometimes the there are no VUS marked or found within an analysis, then no info needs to be uploaded
+            if vus_export == []:
+                logger.info('Patient %s, has no VUS marked/found. Not uploaded to MongoDB' % patient_dn_no)
+            elif vus_export is None:
+                logger.info('Patient %s not uploaded, Alissa database temporarily not available' % patient_dn_no)
+                # TODO send (email) notification of this error, see issue #11 GitHub
+                exit(1)
+            else:
+                upload_to_mongodb(inheritance_analysis, accession_number, analyis_sources, patient_dn_no, vus_export, logger)
+    # if latest date does not retrieve any analyses from Alissa, quit program and try later
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version": 1}, {"lastUpdatedOn": start_time, "version": 1}, True)
+
+
+def upload_to_mongodb(inheritance_analysis, accession_number, analyis_sources, patient_dn_no, vus_export, logger):
+    '''Function, for parsing data to the MongoDB'''
+    # connection with MongoDB and the correct database "vus" and collection "variant"
+    mongo_client = pymongo.MongoClient(config.mongodb_localhost)
+    db = mongo_client.vus.variant
+
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    last_updated_on_mongoDB = db.find_one({"dn_no": patient_dn_no}, {"lastUpdatedOn": 1, "_id": 0})
+    if last_updated_on_mongoDB:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        last_updated_on_mongoDB = last_updated_on_mongoDB['lastUpdatedOn']
+        last_updated_on_Alissa = inheritance_analysis['lastUpdatedOn']
+        if last_updated_on_Alissa == last_updated_on_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no)
+            return
+        elif last_updated_on_Alissa > last_updated_on_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif last_updated_on_Alissa < last_updated_on_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
```

Comment:
This should not happen. When this triggers the data in the VUSualizer is newer then in Alissa. I wanted to add this line, otherwise it will not report anything. The VUSualizer is a new app, so when this issue occurs, it is good to find out what exactly triggered it and we need to discuss with research/lab on how to deal with it or prevent this. However, since I do not know it this is theoretically possible now, I at least want to log it somewhere.
I hope this answers the question?

Reply:
Sounds like a serious issue if the elif would be triggered. Is it possible to change logger.info to a more serious level, such as error?

Reply:
yes! It used to trigger the wrong handler, but now that is removed, so it should work. Will change it 👍